src/diff_file.c

```diff
@@ -46,7 +46,7 @@ static int diff_file_content_init_common(
 {
 	fc->opts_flags = opts ? opts->flags : GIT_DIFF_NORMAL;
 
-	if (opts && opts->max_size >= 0)
+	if (opts && (int64_t)opts->max_size >= 0)
```

Comment:
Mh. This comparison doesn't make sense anymore, as `max_size` is now unsigned. If it becomes negative due to casting then I'd consider this a bug.

Reply:
`max_size` is still signed for backcompat.

Reply:
Yeah, ignore my last comment.  You're right.

Reply:
Fixed.